src/page-modules/assistant/details/trip-section/index.tsx

```diff
@@ -102,30 +121,36 @@ export default function TripSection({
           </TripRow>
         )}
       </div>
+      {interchangeDetails && leg.interchangeTo?.guaranteed && leg.line && (
+        <div className={style.rowContainer}>
+          <DecorationLine color={unknownTransportationColor.backgroundColor} />
+          <TripRow rowLabel={<MonoIcon icon="actions/Interchange" />}>
+            <MessageBox
+              noStatusIcon
+              type="info"
+              message={
+                leg.line.publicCode
+                  ? t(
+                      PageText.Assistant.details.tripSection.interchange(
+                        leg.line.publicCode,
+                        interchangeDetails.publicCode,
+                        interchangeDetails.fromPlace,
+                      ),
+                    )
+                  : t(
+                      PageText.Assistant.details.tripSection.interchangeWithUnknownFromPublicCode(
+                        interchangeDetails.publicCode,
+                        interchangeDetails.fromPlace,
+                      ),
+                    )
+              }
+            />
+          </TripRow>
+        </div>
+      )}
+      {legWaitDetails?.mustWaitForNextLeg && legWaitDetails.waitTime > 30 && (
```

Comment:
These types of checks can also be named. 
```suggestion
      {showWaitSection && (
```

It makes it easier to read components i think

Reply:
Could also potentially pass in `mustWaitForNextLeg` and have the conditional inside `WaitSection`. Reducing the overhead in reading the component. Drawback is that it isn't as that it is optional, but I think the Component Prop signature with `mustWaitForNextLeg | undefined` can be good enough a indicator

Reply:
I like this 👍 I actually think that it is easier to read the component using the named check than passing the `mustWaitForNextLeg` to the component 🤔 I'm up for both changes, but I think the way it is now is good

```tsx
{showInterchangeSection && (
  <InterchangeSection
    interchangeDetails={interchangeDetails}
    publicCode={leg.line?.publicCode}
  />
)}
{showWaitSection && (
  <WaitSection waitTimeInSeconds={legWaitDetails.waitTime} />
)}
```

Reply:
I might have misread as to that `mustWaitForNextLeg` was part of the same `legWaitDetails` object. But yeah, I also think this is fine enough!

Reply:
It actually is, and when you say it like that I guess it makes more sense 😄 I didn't include the `mustWaitForNextLeg` to the `WaitSection` props as it was not needed, but I guess passing the "entire" object and check if it should return something makes sense

Reply:
Then again, we should probably do the same with the `InterchangeSection` and pass `guaranteed` as a prop 🤔

Reply:
That might make sense